flink-formats/flink-orc/src/main/java/org/apache/flink/orc/OrcFilters.java

```diff
@@ -0,0 +1,372 @@
+			// unsupported literal type
+			LOG.debug("Unsupported predicate [{}] cannot be pushed into OrcFileSystemFormatFactory.", callExp);
+			return null;
+		}
+
+		String colName = getColumnName(callExp);
+
+		// fetch literal and ensure it is serializable
+		Object literalObj = getLiteral(callExp).get();
+		Serializable literal;
+		// validate that literal is serializable
+		if (literalObj instanceof Serializable) {
+			literal = (Serializable) literalObj;
+		} else {
+			LOG.warn("Encountered a non-serializable literal of type {}. " +
+							"Cannot push predicate [{}] into OrcFileSystemFormatFactory. " +
+							"This is a bug and should be reported.",
+					literalObj.getClass().getCanonicalName(), callExp);
+			return null;
+		}
+
+		return Tuple3.of(colName, litType, literal);
+	}
+
+	private static Function<CallExpression, OrcSplitReader.Predicate> createIsNullPredicateConverter(){
+		return callExp -> {
+				Tuple2<String, PredicateLeaf.Type> tuple2 = getTuple2Args(callExp);
+
+				if (tuple2 == null){
+					return null;
+				}
+
+				return new OrcSplitReader.IsNull(tuple2.f0, tuple2.f1);
+		};
+	}
+
+	private static Function<CallExpression, OrcSplitReader.Predicate> createIsNotNullPredicateConverter(){
+		return callExp -> {
+
+			Tuple2<String, PredicateLeaf.Type> tuple2 = getTuple2Args(callExp);
+
+			if (tuple2 == null){
+				return null;
+			}
+
+			return new OrcSplitReader.Not(
+					new OrcSplitReader.IsNull(tuple2.f0, tuple2.f1));
+		};
+	}
+
+	private static Function<CallExpression, OrcSplitReader.Predicate> createNotPredicateConverter(){
+		return callExp -> {
+			if (callExp.getChildren().size() != 1) {
+				// not a valid predicate
+				LOG.debug("Unsupported predicate [{}] cannot be pushed into OrcFileSystemFormatFactory.", callExp);
+				return null;
+			}
+
+			OrcSplitReader.Predicate c = toOrcPredicate(callExp.getChildren().get(0));
+			if (c == null) {
+				return null;
+			} else {
+				return new OrcSplitReader.Not(c);
+			}
+		};
+	}
+
+	private static Function<CallExpression, OrcSplitReader.Predicate> createOrPredicateConverter(){
+		return callExp -> {
+			if (callExp.getChildren().size() < 2) {
+				return null;
+			}
+			Expression left = callExp.getChildren().get(0);
+			Expression right = callExp.getChildren().get(1);
+
+			OrcSplitReader.Predicate c1 = toOrcPredicate(left);
+			OrcSplitReader.Predicate c2 = toOrcPredicate(right);
+			if (c1 == null || c2 == null) {
+				return null;
+			} else {
+				return new OrcSplitReader.Or(c1, c2);
+			}
+		};
+	}
+
+	private static Function<CallExpression, OrcSplitReader.Predicate> createEqualsPredicateConverter(){
+		return callExp -> {
+			Tuple3<String, PredicateLeaf.Type, Serializable> tuple3 = getTuple3Args(callExp);
+
+			if (tuple3 == null){
+				return null;
+			}
+			return new OrcSplitReader.Equals(tuple3.f0, tuple3.f1, tuple3.f2);
+		};
+	}
+
+	private static Function<CallExpression, OrcSplitReader.Predicate> createNotEqualsPredicateConverter(){
```

Comment:
You can pass a `TriFunction` to a `convertBinary`:
```
public OrcSplitReader.Predicate convertBinary(CallExpression call,
   TriFunction<String, PredicateLeaf.Type, Serializable, OrcSplitReader.Predicate> func,
   TriFunction<String, PredicateLeaf.Type, Serializable, OrcSplitReader.Predicate> reverseFunc) {
   // move codes from getTuple3Args to here
   // For example, func is LessThan, reverseFunc is greaterThanOrEqual
   return literalOnRight ? func.apply(name, type, literal) : reverseFunc.apply(name, type, literal);
}

```

Reply:
hi @JingsongLi  , This is a good idea , but i think TriFunction<String, PredicateLeaf.Type, Serializable, OrcSplitReader.Predicate> is conflict with Function<CallExpression, OrcSplitReader.Predicate> in FILTERS, We need another static variable, such as BINARY_FILTERS ?

```
private static final ImmutableMap<FunctionDefinition, TriFunction<CallExpression, TriFunction<String, PredicateLeaf.Type, Serializable, OrcSplitReader.Predicate>, TriFunction<String, PredicateLeaf.Type, Serializable, OrcSplitReader.Predicate>, OrcSplitReader.Predicate>> BINARY_FILTERS
```

Reply:
We don't need to change `FILTERS`, for example:
`FILTERS.put(BuiltInFunctionDefinitions.GREATER_THAN, call -> convertBinary(call, OrcSplitReader.GreaterThan::new, OrcSplitReader. LessThanOrEqual::new))`

Reply:
I got it, Thanks a lot